### PR TITLE
Bug/render nodes

### PR DESCRIPTION
**Pull request by simontegg**
Not ready to merge. This renders svg <image>'s by monkey patching `React.DOM` and `SVGDOMPropertyConfig.Properties.`

### CODE REVIEW

File: src/node_modules/graph-view/node-ui.js [hunk 1]
```diff
@@ -24,33 +28,40 @@ var style = {
...
+      })
     )
   },
+
+  href: function () {
+    var d = this.props.datum 
+    debug('d', d)
+    return router.format({ id: this.props.datum.id })
+  },
+
```
Comment (simontegg) [inline src/node_modules/graph-view/node-ui.js]: @ahdinosaur using a onclick handler to router directly as there were problems wrapping an `<image>` with a `Link` (even after my attempt at hacking on `Link`)

getting an error

**Comment by simontegg** [comment]
Ok fixed. I'll tidy this up, rebase and add some tests, but a look over is appreciated

File: src/client.js [hunk 2]
```diff
@@ -8,13 +8,19 @@ var config = require('uiconfig');
 var router = require('router');
 var fetcher = require('fetcher');
 
+require('react-svg-patch');
+
+
 if (config.debug) {
-  localStorage.setItem("debug", "*");
+  localStorage.setItem("debug", "holodex:link");
```
Comment (ahdinosaur) [inline src/client.js]: yeah nah, we should actually change this to `process.env.DEBUG` so you could have done `DEBUG=holodex:link npm run develop` as we should be doing (i'm starting to learn how to use `debug` properly, hehe).

**Comment by ahdinosaur** [comment]
such hacks :fireworks:, but if it works then that's great and looks good to me. also if you're based on master there's no reason to rebase, my mad science `refactor-api` branch is in the stars right now. moving forward, i recommend we link up by each writing tests for our modules' desired interface.

**Comment by simontegg** [comment]
@ahdinosaur  Ok should be ready. Tests are token though.

File: src/node_modules/graph-view/graph.js [hunk 3]
```diff
@@ -52,60 +59,52 @@ module.exports = React.createClass({
...
+          },
+          [ 
+            r.defs(),
+            r.g({className: "edge-group"}, [
+              edges.map(function (edge) {
+                return r(Edge, 
+                  { 
+                    datum: edge, 
+                    actions: actions, 
+                    key: edge.model._values['@id'],
```
Comment (ahdinosaur) [inline src/node_modules/graph-view/graph.js]: `edge.model.getId()`

File: src/node_modules/graph-view/graph.js [hunk 4]
```diff
@@ -52,60 +59,52 @@ module.exports = React.createClass({
...
+                    color: this.props.palleteMap[edge.model._values.relationshipType]
```
Comment (ahdinosaur) [inline src/node_modules/graph-view/graph.js]: `edge.model.relationshipType`

Comment (simontegg) [inline src/node_modules/graph-view/graph.js]: fixed

File: src/node_modules/graph-view/test/test-node-ui.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+var Node = require('graph-view/node-ui')
+var test = require('tape')
+var React = require('react')
+var r = require('r-dom')
+// var ReactTestUtils = require('react-test-utils')
+require('react-svg-patch')
+require('./testdom')('<html><body></body></html>')
```
Comment (ahdinosaur) [inline src/node_modules/graph-view/test/test-node-ui.js]: i'd rather we weren't using jsdom for this, we should be putting our browser-side tests in a module's `test-client` dir so they are run by `npm run test-client`, or directly with `./node_modules/.bin/run-browser src/node_modules/graph-view/test-client -b`, and we can setup our dom with normal dom operations (nice utility modules can be found in the [`npm-dom` namespace](https://github.com/npm-dom)).

Comment (simontegg) [inline src/node_modules/graph-view/test/test-node-ui.js]: cool. Makes sense.

**Comment by ahdinosaur** [comment]
hmm, i'm only seeing a blank screen on firefox and a mostly blank screen with arrows floating around on chrome.

**Comment by simontegg** [comment]
check console. Are they ``'s or `<image>`'s ?

**Comment by simontegg** [comment]
I'm getting arrowheads + square images on firefox (no lines, no css `clip: circle()`

**Comment by ahdinosaur** [comment]
``s. have you changed anything in your local react install that might separate yours from mine?

**Comment by simontegg** [comment]
Yeah its in `getMarkupWrap.js`. Need a little more jigging

**Comment by simontegg** [comment]
patching done. Now fixing tests.

**Comment by simontegg** [comment]
@ahdinosaur ok looking better now

**Comment by ahdinosaur** [comment]
looks good to me, sans some minor nitpicks. :palm_tree:

**Comment by ahdinosaur** [comment]
here's how it looks to me in firefox 35.0.1:
wacky-holodex5

**Comment by ahdinosaur** [comment]
also the amount of hacks you've had to go through to get this to work is astounding, kudos to you @simontegg. :smile_cat:

**Comment by simontegg** [comment]
My firefox is showing the arrowheads and  ``marker-end`? 

Seems firefox doesn't support css `clip-path: circle()`. I feel like that's a separate bug :)

**Comment by simontegg** [comment]
I reckon this can be merged. I can do the firefox issues as a separate bug.

**Comment by ahdinosaur** [comment]
k cool, i'll merge it up, @simontegg can you file the issues?

**Comment by ahdinosaur** [comment]
deployed :tada:

**Pull request merged**